### PR TITLE
Separation of checkAttributeSyntax from semantics in Resource modules

### PR DESCRIPTION
**Pull request by metodej**
- former checkAttributeValue is now separated into checkAttributeSyntax and checkAttributeSemantics in resource attribute modules
- also added test for checks

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_defaultDataQuota.java [hunk 1]
```diff
@@ -119,7 +91,7 @@ public void checkAttributeSemantics(PerunSessionImpl perunSession, Resource reso
 		//Compare DefaultDataQuota with DefaultDataLimit
 		if (quotaNumber.compareTo(BigDecimal.valueOf(0)) == 0) {
 			if (limitNumber.compareTo(BigDecimal.valueOf(0)) != 0) {
-				throw new WrongReferenceAttributeValueException(attribute, attrDefaultDataLimit, resource, null, resource, null, "Try to set unlimited quota, but limit is still " + defaultDataLimitNumber + defaultDataLimitLetter);
+				throw new WrongReferenceAttributeValueException(attribute, attrDefaultDataLimit, resource, null, resource, null, "Try to set unlimited quota, but limit is still " + limitNumber + defaultDataLimitLetter);
 			}
 		} else if (limitNumber.compareTo(BigDecimal.valueOf(0)) != 0 && defaultDataLimitLetter != null && defaultDataQuotaLetter != null) {
```
Comment (balcirakpeter) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_defaultDataQuota.java]: I think you can erase the check if defaultDatalimitLetter is null as you did in the defaultDataLimit module. Moreover, there is the same question on defaultDataQuotaLetter as in the previous module.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_defaultDataLimit.java [hunk 2]
```diff
@@ -121,7 +97,7 @@ public void checkAttributeSemantics(PerunSessionImpl perunSession, Resource reso
 			if (limitNumber.compareTo(BigDecimal.valueOf(0)) != 0) {
 				throw new WrongReferenceAttributeValueException(attribute, attrDefaultDataQuota, resource, null, resource, null, "Try to set limited limit, but there is still set unlimited Quota.");
 			}
-		} else if ((quotaNumber.compareTo(BigDecimal.valueOf(0)) != 0) && (limitNumber.compareTo(BigDecimal.valueOf(0)) != 0) && defaultDataLimitLetter != null && defaultDataQuotaLetter != null) {
+		} else if ((quotaNumber.compareTo(BigDecimal.valueOf(0)) != 0) && (limitNumber.compareTo(BigDecimal.valueOf(0)) != 0) && defaultDataQuotaLetter != null) {
```
Comment (balcirakpeter) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_defaultDataLimit.java]: Why do you need to check if defaultDataQuotaLetter is null when it came from the same method as defaultDataLimitLetter?

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_unixGID_namespace.java [hunk 3]
```diff
@@ -94,26 +94,30 @@ public Attribute fillAttribute(PerunSessionImpl sess, Resource resource, Attribu
...
+				throw new WrongReferenceAttributeValueException(attribute, null, resource, null, "Unix GID must be set");
 			} else {
-				attrValue = (Integer) attribute.getValue();
+				attrValue = attribute.valueAsInteger();
 			}
 
 			//Check if GID is within allowed range
-			sess.getPerunBl().getModulesUtilsBl().checkIfGIDIsWithinRange(sess, attribute);
+			try {
+				sess.getPerunBl().getModulesUtilsBl().checkIfGIDIsWithinRange(sess, attribute);
```
Comment (balcirakpeter) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_unixGID_namespace.java]: I think the method checkIfGIDIsWithinRange should be reworked, so it does not throw WrongAttributeValueException. This Exception is thrown on two places: first if the attribute is null where we usually do not throw Value exception. Second, from method checkAndConvertGIDRanges which check reference attribute so we should catch Value exception and throw the Reference exception.

Comment (metodej) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_unixGID_namespace.java]: We already discussed this method earlier with group module and decided to this solution.

**Comment by metodej** [comment]
Thanks for review, the missed unnecessary checks were removed.

**Pull request merged**